### PR TITLE
BREAKING CHANGE: Migrate AWS Settings from CLI Args to ConfigMap

### PR DESCRIPTION
**Pull request by jonathan-innis**
__BREAKING CHANGE: Integrating this new code will break customers who were using these CLI args__

Fixes # 

**Description**

- Migrates all removed options in https://github.com/aws/karpenter-core/pull/39 to come from `awsettings.FromContext(ctx)`

This is the buddy PR for `karpenter-core`: https://github.com/aws/karpenter-core/pull/39

**How was this change tested?**

* `make battletest`
* `FOCUS=Integration make e2etests`

**Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note
None
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/karpenter/templates/configmap.yaml [hunk 1]
```diff
@@ -10,8 +10,17 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 data:
+  "clusterName": "{{ .Values.clusterName }}"
```
Comment (ellistarn) [inline charts/karpenter/templates/configmap.yaml:13]: If we nested these helm params in a single bucket, we could just wire it through transparently without needing to change this each time.

File: charts/karpenter/values.yaml [hunk 2]
```diff
@@ -138,7 +138,19 @@ clusterName: ""
 # -- Cluster endpoint.
 clusterEndpoint: ""
 aws:
```
Comment (ellistarn) [inline charts/karpenter/values.yaml:140]: thoughts on nesting in something like `global-settings.xyz`

File: charts/karpenter/values.yaml [hunk 3]
```diff
@@ -133,12 +125,34 @@ webhook:
...
-  defaultInstanceProfile: ""
-  # -- The global tags to use on all AWS infrastructure resources (launch templates, instances, SQS queue, etc.)
-  tags:
\ No newline at end of file
+# -- Global Settings to configure Karpenter
+settings:
+  # -- Cluster name.
+  clusterName: ""
+  # -- Cluster endpoint.
+  clusterEndpoint: ""
```
Comment (ellistarn) [inline charts/karpenter/values.yaml:133]: IIUC these are AWS specific.

**Review by ellistarn — DISMISSED** [review]

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**